components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -228,6 +230,32 @@ static esp_err_t esp_websocket_client_error_connection(esp_websocket_client_hand
     return ESP_OK;
 }
 
+static char *http_auth_basic(const char *username, const char *password)
+{
+    int out;
+    char *user_info = NULL;
+    char *digest = NULL;
+    size_t n = 0;
+
+    if (asprintf(&user_info, "%s:%s", username, password) < 0) {
+        return NULL;
+    }
+
+    if (!user_info) {
+        ESP_LOGE(TAG, "No enough memory for user information");
+        return NULL;
+    }
+
+    esp_crypto_base64_encode(NULL, 0, &n, (const unsigned char *)user_info, strlen(user_info));
+    digest = calloc(1, 6 + n + 1);
+    if (digest) {
+        strcpy(digest, "Basic ");
```

Comment:
So we are using hard coded string "Basic" and adding +6 during calculations?  Isn't it better to use constant strings or macros to decrease hard coded values ? Although with configurable macros users can modify it.

Reply:
Thanks for your suggestioning.